.github/workflows/nightly-perf-tests.yml

```diff
@@ -0,0 +1,145 @@
+
+  jib:
+    needs: [ build ]
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ github.ref }}-${{ github.sha }}
+
+      - name: Construct docker image name and tag
+        id: image-name
+        run: echo ::set-output name=name::${{ github.event.repository.name }}:$(git rev-parse --short HEAD)
+
+      - name: Build image locally with jib
+        run: |
+          ./gradlew --build-cache :service:jibDockerBuild \
+            --image=${{ steps.image-name.outputs.name }} \
+            -Djib.console=plain
+
+      - name: Run Trivy vulnerability scanner
+        uses: broadinstitute/dsp-appsec-trivy-action@v1
+        with:
+          image: ${{ steps.image-name.outputs.name }}
+
+  integration-tests:
+    needs: [ build ]
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:13
+        env:
+          POSTGRES_PASSWORD: postgres
+        ports: [ "5432:5432" ]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ github.ref }}-${{ github.sha }}
+
+      - name: Make sure Postgres is ready and init
+        env:
+          PGPASSWORD: postgres
+        run: |
+          pg_isready -h localhost -t 10
+          psql -h localhost -U postgres -f ./common/postgres-init.sql
+
+      - name: Render GitHub Secrets
+        run: echo "${{ secrets.DEV_FIRECLOUD_ACCOUNT_B64 }}" | base64 -d >"./integration/src/main/resources/rendered/user-delegated-sa.json"
+
+      - name: Launch the background process for integration tests
+        run: ./gradlew --build-cache :service:bootRun | tee bootrun.log &
+
+      - name: Prep the integration project
+        run: ./gradlew --build-cache :integration:classes
+
+      - name: Wait for boot run to be ready
+        run: |
+          timeout 60 bash -c 'until echo > /dev/tcp/localhost/8080; do sleep 1; done'
+          resultStatus=$(echo > /dev/tcp/localhost/8080; echo $?)
+          if [[ $resultStatus == 0 ]]; then
+            echo "Server started successfully"
+          else
+            echo "Server did not start successfully. Dumping log file"
+            cat bootrun.log
+            exit 1
+          fi
+
+      - name: Run the integration test suite
+        run: ./gradlew --build-cache :integration:runTest --args="suites/FullIntegration.json /tmp/foo"
```

Comment:
The test suite is different for perf
```suggestion
        run: ./gradlew --build-cache :integration:runTest --args="suites/FullPerf.json /tmp/foo"
```

Reply:
Yes, I agree. I thought we would run both integration and perf tests? The way I wrote it so far would only cover the integration tests.

Reply:
We haven't yet talked about what's in a perf test vs what's in an integration test. One possible approach is to have perf include everything integration does, but with more simultaneous operations.

If we do have a nightly test that runs integration tests, maybe that should use the dev deployment instead of running a local deployment? This can be done using an env var to override the server config setting for the integration tests.